src/Framework/Migrations/Contracts/BatchMigration.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace Give\Framework\Migrations\Contracts;
+
+/**
+ * Extend this class when you need database migration to run in batches.
+ *
+ * @unreleased
+ */
+abstract class BatchMigration extends BaseMigration
+{
+    /**
+     * Get the number of items per batch
+     *
+     * @unreleased
+     */
+    abstract public function getBatchSize(): int;
+
+    /**
+     *
+     * Get the total items count
+     *
+     * @unreleased
+     */
+    abstract public function getItemsCount(): int;
+
+
+    /**
+     *
+     * Get the first and the last item ID for a batch
+     *
+     * @unreleased
+     *
+     * @return array<int, int>
```

Comment:
I don't think this syntax is correct. I believe this is for an associative array where the keys are integers and the values are integers. I believe this is correct:
```suggestion
     * @return array{0: int, 1: int} the first value is the first id and the second value is the last id of a batch
```

Reply:
Resolved https://github.com/impress-org/givewp/pull/7755/commits/144450bf1331e555f2259ff3b80b5179c9867ac7